listener.go

```diff
@@ -22,7 +23,7 @@ type TLSStorage interface {
 }
 
 type TLSFactory interface {
-	Refresh(secret *v1.Secret) (*v1.Secret, error)
+	Renew(secret *v1.Secret) (*v1.Secret, error)
```

Comment:
Would like to understand the reasoning behind changing this exported interface method name. This will have the effect of breaking existing clients when they attempt to upgrade (which may be on purpose, I don't know).

Reply:
The industry standard term for what we are doing here is renewal, not refreshing. See:
https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.7.0-1.pdf section 4.6. Using standard terminology makes it clear what this method is responsible for doing.

Reply:
> The industry standard term for what we are doing here is renewal, not refreshing. See:
> https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.7.0-1.pdf section 4.6. Using standard terminology makes it clear what this method is responsible for doing.

Sorry, I assumed this. I agree it is a better name but are there consequences that we might wish to avoid with such a breaking change in the public API. Yes we own it. I guess what I am asking is, is it only ever "internally" consumed? If so, fine. If not, are we comfortable breaking downstream?

Reply:
I don't see any use of the Renew method in Rancher or K3s; it appears to only be called internal to this module. Given that we don't even have a README in this repo, I somewhat doubt anyone else is using this module at the moment, but I very well could be wrong.